src/qt/walletcontroller.cpp

```diff
@@ -432,3 +432,67 @@ void RestoreWalletActivity::finish()
 
     Q_EMIT finished();
 }
+
+void MigrateWalletActivity::migrate(WalletModel* wallet_model)
+{
+    // Warn the user about migration
+    QMessageBox box(m_parent_widget);
+    box.setWindowTitle(tr("Migrate wallet"));
+    box.setText(tr("Are you sure you wish to migrate the wallet <i>%1</i>?").arg(GUIUtil::HtmlEscape(wallet_model->getDisplayName())));
+    box.setInformativeText(tr("Migrating the wallet will convert this wallet to one or more descriptor wallets. A new wallet backup will need to be made.\n"
+                "If this wallet contains any watchonly those scripts, a new wallet will be created which contains those watchonly scripts.\n"
```

Comment:
I do think this is quite wordy, and maybe more information than is needed. But, no need to reduce the amount of information here.

I know that for some users, they continue to use the Bitcoin Core GUI because their wallet was generated with it in some early version, and they are scared to use anything else but that. So, migrating can be a scary task.

When implementing this functionality for the gui-qml we'd probably make this less intimidating, perhaps as a wizard, and with more interaction/feedback with the process. CC @GBKS @mouxdesign

![Screen Shot 2023-06-22 at 6 55 11 PM](https://github.com/bitcoin-core/gui/assets/23396902/2e0a41f1-eee8-4faa-bbd3-e7832b491b9f)